File: bannerlayout/src/main/java/com/bannerlayout/util/BannerSelectorUtils.java
package com.bannerlayout.util;

import android.graphics.drawable.GradientDrawable;
import android.graphics.drawable.StateListDrawable;

/**
 * by y on 2016/12/28
 */

public class BannerSelectorUtils {


    public static StateListDrawable getDrawableSelector(float enabledRadius,
                                                        int enabledColor,
                                                        float normalRadius,
                                                        int normalColor) {
        StateListDrawable drawable = new StateListDrawable();
        drawable.addState(new int[]{android.R.attr.state_enabled}, getShape(enabledRadius, enabledColor));
        drawable.addState(new int[]{-android.R.attr.state_enabled}, getShape(normalRadius, normalColor));
        return drawable;
    }

    public static GradientDrawable getShape(float radius, int color) {
        GradientDrawable gd = new GradientDrawable();
//        gd.setShape(GradientDrawable.OVAL); The larger the radius, the more round the radius
        gd.setCornerRadius(radius);
        gd.setColor(color);
        return gd;
    }
}

File: bannerlayout/src/main/java/com/bannerlayout/util/BannerSelectorUtils.kt
package com.bannerlayout.util

import android.graphics.drawable.GradientDrawable
import android.graphics.drawable.StateListDrawable

/**
 * by y on 2016/12/28
 */

object BannerSelectorUtils {


    fun getDrawableSelector(enabledRadius: Float,
                            enabledColor: Int,
                            normalRadius: Float,
                            normalColor: Int): StateListDrawable {
        val drawable = StateListDrawable()
        drawable.addState(intArrayOf(android.R.attr.state_enabled), getShape(enabledRadius, enabledColor))
        drawable.addState(intArrayOf(-android.R.attr.state_enabled), getShape(normalRadius, normalColor))
        return drawable
    }

    fun getShape(radius: Float, color: Int): GradientDrawable {
        val gd = GradientDrawable()
        gd.cornerRadius = radius
        gd.setColor(color)
        return gd
    }
}
